pkg/azuredisk/controllerserver.go

```diff
@@ -839,6 +845,35 @@ func (d *Driver) getSnapshotByID(ctx context.Context, resourceGroup, snapshotID,
 	return generateCSISnapshot(sourceVolumeID, &snapshot)
 }
 
+// GetSourceDiskSize returns: disk size, error
+func (d *Driver) GetSourceDiskSize(resourceGroup, diskName string) (*int32, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	return d.GetSourceDiskSizeHelper(ctx, resourceGroup, diskName)
+}
+
+// GetSourceDiskSizeHelper is a helper function for GetSourceDiskSize to support recursive search for  and it returns: disk size, error
```

Comment:
recursive search for sourceDisk?

Reply:
It is to prevent (https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/657#discussion_r552913192).

Reply:
Oh! I meant that the comment seems incomplete. " GetSourceDiskSize to support recursive search for  and it returns". Should it be recursive search for sourceDisk instead?

Reply:
Oh, thanks for pointing that out. I must have deleted that part in process of editing comments.